lib/src/rules/prefer_const_constructors_in_immutables.dart

```diff
@@ -120,20 +120,22 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   bool _hasImmutableAnnotation(ClassElement clazz) {
     var selfAndInheritedClasses = _getSelfAndInheritedClasses(clazz);
-    var selfAndInheritedAnnotations =
-        selfAndInheritedClasses.expand((c) => c.metadata).map((m) => m.element);
-    return selfAndInheritedAnnotations.any(_isImmutable);
+    for (var cls in selfAndInheritedClasses) {
+      if (cls.metadata.any((m) => _isImmutable(m.element))) return true;
+    }
+    return false;
   }
 
   bool _hasMixin(ClassElement clazz) => clazz.mixins.isNotEmpty;
 
-  static Iterable<ClassElement> _getSelfAndInheritedClasses(
-      ClassElement self) sync* {
+  static Iterable<ClassElement> _getSelfAndInheritedClasses(ClassElement self) {
```

Comment:
You can change the return type to `List`, so that consumers will know that it is not necessary to call `toList()` in case if they iterate multiple times. Or not, if you know that the4 client does exactly one iteration.

Reply:
The name is not very specific. It says "inherited", but actually wants "super".

Reply:
Updated.  Thanks!